### DISCUSSION TITLE
Aidex integration - misses readings 2-4 times per hour (but Aidex app gets them)

### DISCUSSION THREAD

**Discussion by nathanscrivener** [NightscoutFoundation/xDrip#3194] [Bug reports]
Hi, we have recently changed into Aidex Lite CGM. The integration with xDrip is working but it sporadically misses readings. These are showing in the Aidex app.

xDrip version: 23ccbd1-2023.07.16
Phone: Samsung A52, Model number SM-A525F/DS
Hardware/data source set to GlucoRX/Aidex

![IMG_2758](https://github.com/NightscoutFoundation/xDrip/assets/35670110/081fe570-4cf7-4bb3-b34c-d06e0c86d3ff)
![IMG_2757](https://github.com/NightscoutFoundation/xDrip/assets/35670110/f7e75c83-24d2-4fc3-b637-485db6af1f6e)

![IMG_2759](https://github.com/NightscoutFoundation/xDrip/assets/35670110/1290548f-bcd5-4fc3-b5a1-56f1d1b8497a)

![image](https://github.com/NightscoutFoundation/xDrip/assets/35670110/610b461f-018c-45df-8d5a-598147b753ba)

**Comment by jamorham** [comment]
@andyrozman any ideas?

**Comment by andyrozman** [comment]
We only get transmited current values. So if phone misses some values Aidex application would still be able to display them (from history), but only values that are broadcasted are current values... 

Samsung phone have notirously bad BT implementation, so lossing reading would be dirrect result of that.

**Reply by nathanscrivener** [reply]
Hi Andy, I don't think it's that. When the reading is missed in xDrip you can open the Aidex application and see the reading there.

**Comment by andyrozman** [comment]
Yes, because Aidex application can see the readings from history. If application discovers that readings were missed it can request history from transmiter, so it then displays all entries from past. But Aidex application will broadcast ONLY latest value (when that value is received). Values from that history reading won't be transmitted. 

I used Samsung in the past (years ago) myself, and I always had problems with it (with AAPS and Xdrip). This is why there was this option in Xdrip introduced about "Buggy Samsung".  You need to have Location and BT always on for Aidex to work correctly and check if you have disabled all energy saving for both apps (Aidex and Xdrip) and for BT and Location (if your phone has option to Energy save thoose 2).
